components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml

```diff
@@ -694,32 +668,30 @@
           dataObj.collaborators = {"collaborators" : collaboratorsArray};
         }
 
-        dataObjString = JSON.stringify(dataObj);
-        _this._ajaxCallsRemaining = 0;
+        var hasChanged = dataObj.owner || dataObj.visibility || dataObj.collaborators;
+        if (!hasChanged &amp;&amp; !_this.collaboratorRemoved) {
+          _this._dialog.closeDialog();
+        }
 
-        
         // if something was changed we update permissions with PATCH
-        if (dataObjString.length &gt; 3) {
-          _this._updatePermissions(dataObjString);
-        } else {
+        if (hasChanged) {
+          _this._updatePermissions(dataObj, collaboratorsArray);
```

Comment:
Nevermind, I see. I think I would have sacrificed performance for code simplicity and would have always made a separate REST call for collaborators. But this works.